mission_control/test/load_execute_mission.py

```diff
@@ -99,49 +77,35 @@ def setUp(self):
 
     def callback_mission_execute_state(self, msg):
         self.report_execute_mission = msg
-        rospy.loginfo(msg)
-
-    def callback_report_mission(self, msg):
-        self.report_mission = msg
 
     def callback_mission_load_state(self, msg):
         self.mission_load_state = msg.load_state
 
-    def callback_publish_fixed_rudder_goal(self, msg):
-        self.fixed_rudder_goal = [msg.depth,
-                                  msg.rudder,
-                                  msg.speed_knots,
-                                  msg.ena_mask]
-
-    def test_mission_with_fixed_rudder_behavioral(self):
-
+    def load_mission(self, mission_name):
+        # Simulate Jaus Ros Bridge sending a Load Command and wait for
+        # response that the mission was loaded correctly
         self.mission_load_state_flag = False
         mission_to_load = LoadMission()
-        mission_to_load.mission_file_full_path = self.dir_path + "test_missions/fixed_rudder_mission_test.xml"
-        self.simulated_mission_control_load_mission_pub.publish(mission_to_load)
+        mission_to_load.mission_file_full_path = os.path.dirname(
+            os.path.abspath(__file__)) + '/test_files/test_missions/' + mission_name
+        self.simulated_mission_control_load_mission_pub.publish(
+            mission_to_load)
 
-        def load_mission():
+        def load_mission_state():
             return self.mission_load_state == ReportLoadMissionState.SUCCESS
-        self.assertTrue(wait_for(load_mission),
-                        msg='Mission control must report SUCCESS')
-        rospy.loginfo("Mission Loaded")
+        self.wait_for(load_mission_state)
 
+    def execute_mission(self, mission_id=1):
+        # Simulate Jaus Ros Bridge sending an Execute Command
         mission_to_execute = ExecuteMission()
-        mission_to_execute.mission_id = 1
+        mission_to_execute.mission_id = mission_id
         self.simulated_mission_control_execute_mission_pub.publish(
             mission_to_execute)
-        rospy.loginfo("execute msg")
-
-        def fixed_rudder_goals_are_setted():
-            return self.fixed_rudder_goal == [1.0, 2.0, 3.0, 7]
-        self.assertTrue(wait_for(fixed_rudder_goals_are_setted),
-                        msg='Mission control must publish goals')
-
-        def success_mission_status_is_reported():
-            return self.report_execute_mission.execute_mission_state == ReportExecuteMissionState.COMPLETE
-        self.assertTrue(wait_for(success_mission_status_is_reported), 
-            msg='Mission control must report SUCCESS')
 
-if __name__ == "__main__":
-    rostest.rosrun('mission_control', 'mission_control_test_fixed_rudder_behavioral',
-                   TestFixedRudderBehavior)
+    def wait_for(self, predicate, period=1):
```

Comment:
@patapufate consider keeping this a free function.

Reply:
Done